diff --git a/include/AntTweakBar.h b/include/AntTweakBar.h
index ca2ed24..aed08e0 100755
--- a/include/AntTweakBar.h
+++ b/include/AntTweakBar.h
@@ -43,7 +43,7 @@
 // ----------------------------------------------------------------------------
 
 #if (defined(_WIN32) || defined(_WIN64)) && !defined(TW_STATIC)
-#   define TW_CALL          __stdcall
+#   define TW_CALL          //__stdcall
 #   define TW_CDECL_CALL    __cdecl
 #   define TW_EXPORT_API    __declspec(dllexport)
 #   define TW_IMPORT_API    __declspec(dllimport)
