diff --git a/examples/glfw.h b/examples/glfw.h
index e669872..4a2526f 100755
--- a/examples/glfw.h
+++ b/examples/glfw.h
@@ -130,8 +130,8 @@ extern "C" {
 
  // We are building a Win32 DLL
  #define GLFWAPI      __declspec(dllexport)
- #define GLFWAPIENTRY __stdcall
- #define GLFWCALL     __stdcall
+ #define GLFWAPIENTRY //__stdcall
+ #define GLFWCALL     //__stdcall
 
 #elif defined(_WIN32) && defined(GLFW_DLL)
 
@@ -141,8 +141,8 @@ extern "C" {
  #else
   #define GLFWAPI      __declspec(dllimport)
  #endif
- #define GLFWAPIENTRY __stdcall
- #define GLFWCALL     __stdcall
+ #define GLFWAPIENTRY //__stdcall
+ #define GLFWCALL     //__stdcall
 
 #else
 
diff --git a/include/AntTweakBar.h b/include/AntTweakBar.h
index 41bb77c..b255eb1 100755
--- a/include/AntTweakBar.h
+++ b/include/AntTweakBar.h
@@ -43,7 +43,7 @@
 // ----------------------------------------------------------------------------
 
 #if defined(_WIN32) || defined(_WIN64)
-#   define TW_CALL          __stdcall
+#   define TW_CALL          //__stdcall
 #   define TW_CDECL_CALL    __cdecl
 #   define TW_EXPORT_API    __declspec(dllexport)
 #   define TW_IMPORT_API    __declspec(dllimport)
diff --git a/src/TwBar.cpp b/src/TwBar.cpp
index 6978201..328b128 100755
--- a/src/TwBar.cpp
+++ b/src/TwBar.cpp
@@ -3488,14 +3488,14 @@ int CTwBar::SetAttrib(int _AttribID, const char *_Value)
             {
                 g_TwMgr->m_BarAlwaysOnTop = m_Name;
                 if( g_TwMgr->m_BarAlwaysOnBottom.length()>0 && strcmp(g_TwMgr->m_BarAlwaysOnBottom.c_str(), m_Name.c_str())==0 )
-                    g_TwMgr->m_BarAlwaysOnBottom.clear();
+		  g_TwMgr->m_BarAlwaysOnBottom.erase( g_TwMgr->m_BarAlwaysOnBottom.begin(), g_TwMgr->m_BarAlwaysOnBottom.end() );
                 TwSetTopBar(this);
                 return 1;
             }
             else if( _stricmp(_Value, "0")==0 || _stricmp(_Value, "false")==0 )
             {
                 if( g_TwMgr->m_BarAlwaysOnTop.length()>0 && strcmp(g_TwMgr->m_BarAlwaysOnTop.c_str(), m_Name.c_str())==0 )
-                    g_TwMgr->m_BarAlwaysOnTop.clear();
+		  g_TwMgr->m_BarAlwaysOnTop.erase( g_TwMgr->m_BarAlwaysOnTop.begin(), g_TwMgr->m_BarAlwaysOnTop.end() );
                 return 1;
             }
             else
@@ -3516,14 +3516,14 @@ int CTwBar::SetAttrib(int _AttribID, const char *_Value)
             {
                 g_TwMgr->m_BarAlwaysOnBottom = m_Name;
                 if( g_TwMgr->m_BarAlwaysOnTop.length()>0 && strcmp(g_TwMgr->m_BarAlwaysOnTop.c_str(), m_Name.c_str())==0 )
-                    g_TwMgr->m_BarAlwaysOnTop.clear();
+		  g_TwMgr->m_BarAlwaysOnTop.erase( g_TwMgr->m_BarAlwaysOnTop.begin(), g_TwMgr->m_BarAlwaysOnTop.end() );
                 TwSetBottomBar(this);
                 return 1;
             }
             else if( _stricmp(_Value, "0")==0 || _stricmp(_Value, "false")==0 )
             {
                 if( g_TwMgr->m_BarAlwaysOnBottom.length()>0 && strcmp(g_TwMgr->m_BarAlwaysOnBottom.c_str(), m_Name.c_str())==0 )
-                    g_TwMgr->m_BarAlwaysOnBottom.clear();
+		  g_TwMgr->m_BarAlwaysOnBottom.erase( g_TwMgr->m_BarAlwaysOnBottom.begin(), g_TwMgr->m_BarAlwaysOnBottom.end() );
                 return 1;
             }
             else
diff --git a/src/TwFonts.cpp b/src/TwFonts.cpp
index 19ac238..8efdd7a 100755
--- a/src/TwFonts.cpp
+++ b/src/TwFonts.cpp
@@ -11,9 +11,10 @@
 #include "TwPrecomp.h"
 #include "TwMgr.h"
 #include "TwFonts.h"
+#include <string.h>
 
 // Fedora patch: memset()
-using std::memset;
+//using std::memset;
 
 //  ---------------------------------------------------------------------------
 
