diff --git a/src/win32_opengl.c b/src/win32_opengl.c
index d6e1b49..8477cbb 100644
--- a/src/win32_opengl.c
+++ b/src/win32_opengl.c
@@ -38,7 +38,7 @@
 // Thread local storage attribute macro
 //========================================================================
 #if defined(_MSC_VER)
- #define _GLFW_TLS __declspec(thread)
+#define _GLFW_TLS //__declspec(thread)
 #elif defined(__GNUC__)
  #define _GLFW_TLS __thread
 #else
