diff --git a/src/win32fixes.c b/src/win32fixes.c
index eb30cf3..4a0d646 100644
--- a/src/win32fixes.c
+++ b/src/win32fixes.c
@@ -468,13 +468,20 @@ int gettimeofday(struct timeval *tv, struct timezone *tz)
   return 0;
 }
 
+#if _MSC_VER > 1500
 static _locale_t clocale = NULL;
+#endif
+
 double wstrtod(const char *nptr, char **eptr) {
     double d;
     char *leptr;
+#if _MSC_VER > 1500
     if (clocale == NULL)
         clocale = _create_locale(LC_ALL, "C");
     d = _strtod_l(nptr, &leptr, clocale);
+#else
+    d = strtod(nptr, &leptr);
+#endif
     /* if 0, check if input was inf */
     if (d == 0 && nptr == leptr) {
         int neg = 0;
