Index: atomic.h
===================================================================
--- atomic.h	(revision 1457)
+++ atomic.h	(working copy)
@@ -75,7 +75,7 @@
 	return __sync_fetch_and_add (atomic, val);
 #elif defined( _WIN32 )
 /* Windows intrinsic */
-	return _InterlockedExchangeAdd ((volatile LONG*)atomic, val);
+	return InterlockedExchangeAdd ((volatile LONG*)atomic, val);
 #else
 #	error "No supported atomic operations for this platform."
 #endif
@@ -109,7 +109,7 @@
 /* interchangable with __sync_fetch_and_add () */
 	__sync_add_and_fetch (atomic, val);
 #elif defined( _WIN32 )
-	_InterlockedExchangeAdd ((volatile LONG*)atomic, val);
+	InterlockedExchangeAdd ((volatile LONG*)atomic, val);
 #endif
 }
 
@@ -139,7 +139,7 @@
 #elif defined( __GNUC__ ) && ( __GNUC__ * 100 + __GNUC_MINOR__ >= 401 )
 	pgm_atomic_add32 (atomic, 1);
 #elif defined( _WIN32 )
-	_InterlockedIncrement ((volatile LONG*)atomic);
+	InterlockedIncrement ((volatile LONG*)atomic);
 #endif
 }
 
@@ -170,7 +170,7 @@
 #elif defined( __GNUC__ ) && ( __GNUC__ * 100 + __GNUC_MINOR__ >= 401 )
 	pgm_atomic_add32 (atomic, (uint32_t)-1);
 #elif defined( _WIN32 )
-	_InterlockedDecrement ((volatile LONG*)atomic);
+	InterlockedDecrement ((volatile LONG*)atomic);
 #endif
 }
 
