diff --git a/builds/msvc/libzmq/libzmq.vcproj b/builds/msvc/libzmq/libzmq.vcproj
index 464e052..5878ea2 100644
--- a/builds/msvc/libzmq/libzmq.vcproj
+++ b/builds/msvc/libzmq/libzmq.vcproj
@@ -451,11 +451,11 @@
 				>
 			</File>
 			<File
-				RelativePath="..\..\..\src\xrep.cpp"
+				RelativePath="..\..\..\src\router.cpp"
 				>
 			</File>
 			<File
-				RelativePath="..\..\..\src\xreq.cpp"
+				RelativePath="..\..\..\src\dealer.cpp"
 				>
 			</File>
 			<File
@@ -470,6 +470,10 @@
 				RelativePath="..\..\..\src\zmq_utils.cpp"
 				>
 			</File>
+			<File
+				RelativePath="..\..\..\src\device.cpp"
+				>
+			</File>
 		</Filter>
 		<Filter
 			Name="Header Files"
